### PR TITLE
Performance tests for loading of plugins and different feature combinations

### PR DESCRIPTION
**Pull request by oliveromahony**
### Proposed changes

Trying to move code around so it's easier to test the main.go functionality in terms of performance. Added performance tests around plugin combinations. 

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [`CONTRIBUTING`](https://github.com/nginx/agent/blob/main/docs/CONTRIBUTING.md) document
- [x] I have run ```make install-tools``` and have attached any dependency changes to this pull request
- [x] If applicable, I have added tests that prove my fix is effective or that my feature works
- [x] If applicable, I have checked that any relevant tests pass after adding my changes
- [ ] If applicable, I have updated any relevant documentation ([`README.md`](https://github.com/nginx/agent/blob/main/README.md))
- [ ] If applicable, I have tested my cross-platform changes on Ubuntu 22, Redhat 8, SUSE 15 and FreeBSD 13

### CODE REVIEW

File: src/plugins/common.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+		if loadedConfig.IsFeatureEnabled(agent_config.FeatureFileWatcher) {
+			corePlugins = append(corePlugins,
+				NewFileWatcher(loadedConfig, env),
+				NewFileWatchThrottle(),
+			)
+		}
+	}
+
+	if loadedConfig.IsFeatureEnabled(agent_config.FeatureMetrics) || loadedConfig.IsFeatureEnabled(agent_config.FeatureMetricsSender) && reporter != nil {
```
Comment (Dean-Coakley) [inline src/plugins/common.go:34]: Lets ensure the fix from https://github.com/nginx/agent/pull/465 is carried over:
```suggestion
	if (loadedConfig.IsFeatureEnabled(agent_config.FeatureMetrics) || loadedConfig.IsFeatureEnabled(agent_config.FeatureMetricsSender)) && reporter != nil {

```

**Review by Dean-Coakley — APPROVED** [review]

File: sdk/agent/events/meta.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+func NewAgentEventMeta(
+	module, version, pid, message, hostname, systemUuid, instanceGroup string,
+	tags []string,
+) *AgentEventMeta {
+	return &AgentEventMeta{
+		module:        module,
+		version:       version,
+		pid:           pid,
+		message:       message,
```
Comment (dhurley) [inline sdk/agent/events/meta.go:41]: No need for a message field as it should always be `nginx-agent`

File: src/plugins/events.go [hunk 3]
```diff
@@ -26,8 +26,6 @@ import (
 )
 
 const (
-	MODULE = "NGINX-AGENT"
-
 	AGENT_START_MESSAGE             = "nginx-agent %s started on %s with pid %s"
 	AGENT_STOP_MESSAGE              = "nginx-agent %s (pid: %s) stopped on %s"
```
Comment (dhurley) [inline src/plugins/events.go:30]: AGENT_STOP_MESSAGE isn't used anymore so can be removed

File: sdk/agent/events/meta.go (hunk 2 above)
Comment (oliveromahony) [reply sdk/agent/events/meta.go:41]: in the Agent but is someone uses the SDK they can send their own messages

**Review by dhurley — APPROVED** [review]

**Pull request merged**